docs/upgrade.html

```diff
@@ -76,6 +76,10 @@
         <code>org.apache.kafka.common.Metric#metricValue()</code> can be used to retrieve numeric and non-numeric metric values.</li>
     <li>Mx4j will only be enabled if the system property <code>kafka_mx4jenable</code> is set to <code>true</code>. Due to a logic
         inversion bug, it was previously enabled by default and disabled if <code>kafka_mx4jenable</code> was set to <code>true</code>.</li>
+    <li>When using an Authorizer and a user doesn't have Describe authorization on a topic, the broker
+        will return TOPIC_AUTHORIZATION_FAILED errors to requests irrespective of topic existence on broker.
+        If the user have Describe authorization and the topic doesn't exists, then the UNKNOWN_TOPIC_OR_PARTITION
+        error code will be returned. </li>
```

Comment:
Are there any compatibility implications for this? Would a KIP be needed?

Reply:
The main impact is that the client will fail faster in some cases because we will now get an unambiguous error code. I can't think of a situation where you'd depend on the current behavior... Maybe someone might expect the client to retry internally if it is started prior to granting of permissions?

Reply:
Yeah, seems unlikely, just wanted to double-check. This would be a nice complement to the authentication improvements by @rajinisivaram.

Reply:
@rajinisivaram, any concerns?

Reply:
No concerns, sounds like the right behaviour.